webapp/src/ts/modules/analytics/analytics-target-aggregates.component.ts

```diff
@@ -1,110 +1,119 @@
+    this.globalActions.clearSidebarFilter();
   }
 
   private subscribeToStore() {
     const selectorsSubscription = combineLatest(
       this.store.select(Selectors.getTargetAggregates),
       this.store.select(Selectors.getSelectedTargetAggregate),
       this.store.select(Selectors.getTargetAggregatesError),
-    )
-      .subscribe(([
-        aggregates,
-        selected,
-        error
-      ]) => {
-        this.aggregates = aggregates;
-        this.selected = selected;
-        this.error = error;
-      });
+    ).subscribe(([
+      aggregates,
+      selected,
+      error,
+    ]) => {
+      this.aggregates = aggregates;
+      this.selected = selected;
+      this.error = error;
+    });
     this.subscriptions.add(selectorsSubscription);
   }
 
-  getTargetAggregates(userFacilityId?) {
-    return this.targetAggregatesService
-      .isEnabled()
-      .then(enabled => {
-        this.enabled = enabled;
-
-        if (!this.enabled) {
-          return;
-        }
+  async getTargetAggregates(userFacility) {
+    try {
+      const aggregates = this.enabled ? await this.targetAggregatesService.getAggregates(userFacility?._id) : [];
+      if (this.sidebarFilter.hasFacilityFilter) {
+        aggregates?.forEach((aggregate) => aggregate.facility = userFacility?.name);
+      }
+      this.targetAggregatesActions.setTargetAggregates(aggregates);
+      this.targetAggregatesActions.setTargetAggregatesLoaded(true);
 
-        return this.targetAggregatesService.getAggregates(userFacilityId);
-      })
-      .then(aggregates => {
-        this.targetAggregatesActions.setTargetAggregates(aggregates);
-        this.targetAggregatesActions.setTargetAggregatesLoaded(true);
-      })
-      .catch(err => {
-        console.error('Error getting aggregate targets', err);
-        this.targetAggregatesActions.setTargetAggregatesError(err);
-      })
-      .finally(() => {
-        this.loading = false;
-        this.trackPerformance?.stop({
-          name: [ 'analytics', 'target_aggregates', 'load' ].join(':'),
-          recordApdex: true,
-        });
+    } catch (error) {
+      console.error('Error getting aggregate targets', error);
+      this.targetAggregatesActions.setTargetAggregatesError(error);
+    } finally {
+      this.loading = false;
+      this.trackPerformance?.stop({
+        name: [ 'analytics', 'target_aggregates', 'load' ].join(':'),
+        recordApdex: true,
       });
+    }
   }
 
   private subscribeSidebarFilter() {
-    if (!this.useSidebarFilter) {
-      return;
-    }
-
     const subscription = this.store
       .select(Selectors.getSidebarFilter)
-      .subscribe((filterState) => this.isSidebarFilterOpen = filterState?.isOpen ?? false);
+      .subscribe(sidebarFilter => {
+        this.sidebarFilter = sidebarFilter;
+        if (!this.initialLoad && this.sidebarFilter?.defaultFilters) {
+          this.initialLoad = this.getTargetAggregates(this.sidebarFilter.defaultFilters.facility);
```

Comment:
I am probably missing something here, but this feels like an odd case to use a subscription.  We only want to set the target aggregates to the default facility once (when we first load into the target aggregates component).  But, we are subscribed to `getSidebarFilter`, so this will get called each time something changes with the filter.  So, you put in the `initialLoad` variable so we only run this once...  For something that should only happen once, it seems like we do not need a subscription.  

Could we maybe just move the `getTargetAggregates` call down into `setDefaultFilters` and get rid of this function (and `initialLoad`) altogether?

Reply:
Yes, totally! That's how it started, but later made that setDefaultFilters function and didn't see this simplification. 

We're adding another filter (by period) with a default value, so this is perfect. Thanks!